src/pages/Compare/index.spec.tsx

```diff
@@ -127,9 +152,192 @@ describe(__filename, () => {
     render({
       _fetchVersion,
       store,
+      ...getRouteParams({
+        addonId,
+        baseVersionId: baseVersion.id,
+        headVersionId: headVersion.id,
+      }),
+    });
+
+    expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(_fetchVersion).toHaveBeenCalledWith({
+      addonId,
+      versionId: baseVersion.id,
+    });
+  });
+
+  it('redirects to a new compare url when the "old" version is newer than the "new" version', () => {
+    const addonId = 123456;
+    const baseVersion = fakeVersion;
+    const headVersion = { ...fakeVersion, id: baseVersion.id - 1 };
+    const lang = 'es';
+
+    const store = configureStore();
+    const dispatch = spyOn(store, 'dispatch');
+    const fakeThunk = createFakeThunk();
+    const _fetchVersion = fakeThunk.createThunk;
+    const history = createFakeHistory();
+    const push = spyOn(history, 'push');
+
+    render({
+      _fetchVersion,
+      store,
+      ...getRouteParams({
+        addonId,
+        baseVersionId: baseVersion.id,
+        headVersionId: headVersion.id,
+      }),
+      history,
+      lang,
+    });
+
+    expect(push).toHaveBeenCalledWith(
+      `/${lang}/compare/${addonId}/versions/${headVersion.id}...${
+        baseVersion.id
+      }/`,
+    );
+    expect(dispatch).not.toHaveBeenCalled();
+  });
+
+  it('does not dispatch fetchVersion() on update if no parameter has changed', () => {
+    const addonId = 123456;
+    const baseVersion = fakeVersion;
+    const headVersion = { ...fakeVersion, id: baseVersion.id + 1 };
+
+    const store = configureStore();
+    const fakeThunk = createFakeThunk();
+    const dispatch = spyOn(store, 'dispatch');
+
+    const root = render({
+      _fetchVersion: fakeThunk.createThunk,
       addonId: String(addonId),
       baseVersionId: String(baseVersion.id),
       headVersionId: String(headVersion.id),
+      store,
+    });
+
+    dispatch.mockClear();
+    root.setProps({
+      match: {
+        params: {
+          addonId: String(addonId),
+          baseVersionId: String(baseVersion.id),
+          headVersionId: String(headVersion.id),
+        },
+      },
+    });
+
+    expect(dispatch).not.toHaveBeenCalled();
+  });
+
+  it('dispatches fetchVersion() on update if base version is different', () => {
+    const addonId = 123456;
+    const baseVersion = fakeVersion;
+    const headVersion = { ...fakeVersion, id: baseVersion.id + 1 };
```

Comment:
`headVersionId` doesn't seem relevant to this test so it can be moved to a default in the helpers (if possible).